docs/guide/search/elastic.md

```diff
@@ -0,0 +1,558 @@
+                ssl:
+                    cert:
+                        path: '/path/to/cert.pem'
+                        pass: ~
+                    cert_key:
+                        path: '/path/to/cert-key'
+                        pass: ~
+    ```
+
+    After you have configured SLL, you can still disable it, for example` when the certificates
+    expire, or you are migrating to a new set of certificates.
+    To do this, pass the following setting under the `ssl` key:
+
+    ``` yaml
+    verification: false
+    ```
+
+    For more information, see [Elasticsearch: SSL Encyption](https://www.elastic.co/guide/en/elasticsearch/client/php-api/7.x/security.html#_ssl_encryption_2).
+
+    #### Enable debugging
+
+    In a staging environment, you can log messages about the status of communication with Elasticsearch.
+    You can then use Symfony Profiler to review the logs.
+
+    By default, debugging is disabled. To enable debugging, you can toggle either of the following two settings:
+
+    ``` yaml
+    <connection_name>:
+        # ...
+        debug:	<true/false>
+        trace:	<true/false>
+    ```
+
+    - `debug` logs basic information about a request, such as request status and time.
+    - `trace` logs additional information, such as steps to reproduce an exact copy of a query.
+
+    !!! tip
+
+        Make sure that you disable debugging in a production environment.
+
+    ### Configure field type mapping templates
+
+    Before you can re-index the eZ Platform data, so that Elasticsearch can search through its contents, you must define an index template.
+    Templates instruct Elasticsearch to recognize eZ Platform Fields as specific data types, based on, for example, a field name.
+    They help you prevent Elasticsearch from using the dynamic field mapping feature to create type mappings automatically.
+    You can create several field type mapping templates for each index, for example, to define settings that are specific for different languages.
+    When you establish a relationship between a field mapping template and a connection, you can apply several templates, too.
+
+    #### Define a template
+
+    To define a field mapping template, you must provide a number of settings under the `index_templates` key.
+    The structure of the template is as follows:
+
+    ``` yaml
+    ez_platform_elastic_search_engine:
+        # ...
+        index_templates:
+            <index_template_name>:
+                patterns:
+                # ...
+                settings:
+                # ...
+                mappings:
+                # ...
+    ```
+
+    Set a unique name for the template and configure the following keys:
+
+    - `patterns` - A list of wildcards that Elasticsearch uses to match the field mapping template to an index.
+    If you use the `_location_*` wildcard as a pattern, the settings and mappings listed below apply
+    to all indexes whose names are prefixed with the word "location".
```

Comment:
We are missing here documentation for index naming convention. User will not be able to to provide a valid index name pattern without it.